retry/retry.go

```diff
@@ -138,7 +146,7 @@ func (t ticker) StopChan() <-chan struct{} {
 }
 
 func (t ticker) Stop() {
-	t.s <- struct{}{}
+	close(t.s)
```

Comment:
I wonder if that should be wrapped with `sync.Once` to allow several `Stop()` calls like `time.Ticker` allows

Reply:
I think `sync.Once` is pretty heavyweight, I don't think we even ever used `.Stop()` in our codebase (never saw it myself, but still could be used somewhere). 

Probably even `.Stop()` is no longer required given that we have a context which can be canceled (but that is not backwards compatible).

Reply:
so, I would rather not do it unless we see a big use case for `.Stop()`

Reply:
Ok, makes sense